### DISCUSSION TITLE
Intune Deployment

### DISCUSSION THREAD

**Discussion by mtengmo** [Romanitho/Winget-AutoUpdate#88] [General]
How do I deploy it with Intune? 
Make Win32 file of the files, and run the .\winget-install-and-update.ps1 -Silent -UpdatesAtLogon -UpdatesInterval Weekly ?
What kind of detection rules to use?

**Comment by mtengmo** [comment]
C:\ProgramData\Winget-AutoUpdate is empty after installation, but I got the scehduled task created.
Running as a Win32 app (Intunefile) and calling install.bat. Could it be some path lookup issue between scripts?

**Reply by Romanitho** [reply]
I'm on it. It seems to be something with registry 32/64

**Reply by Romanitho** [reply]
Ok, it's definitly due to 32/64 Win32 intunewin app (explainations [here](https://www.anoopcnair.com/intune-win32-app-deploy-system32-vs-syswow64/)). How to use it now:  

- Install cmd (for example with specific notification level):  
   `"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -noprofile -executionpolicy bypass -file "Winget-AutoUpdate-Install.ps1" -Silent -NotificationLevel SuccessOnly`

- Uninstall cmd:  
   `"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -noprofile -executionpolicy bypass -file "Winget-AutoUpdate-Install.ps1" -Uninstall`

- Detection method:  
Key path: `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate`
![image](https://user-images.githubusercontent.com/96626929/171011608-bc369180-64e9-4e0e-abd7-66e284eda10d.png)

**Reply by mtengmo** [reply]
Thanks, I will give it a try.

**Reply by Romanitho** [reply]
It's really important to use 'sysnative' virtual path because Win32 apps run under 32-bit, so we need to force intune to run 64-bit powershell. More info [here](https://www.samlogic.net/articles/sysnative-folder-64-bit-windows.htm)

**Reply by mtengmo** [reply]
Working, perfect. 

But I got two installation of same application. Is it my mistake? 
![image](https://user-images.githubusercontent.com/17195619/171168886-6d4aeae0-76d7-4a52-b20e-2de8ad8f6ae9.png)

And another question, if I want to exclude more apps, do I need to do a new package and reinstall it again?

Thanks, Magnus

**Reply by Romanitho** [reply]
You have two apps because your last try of installation was donc with 32bit context. So it has created uninstall entries in 32 registry as well. I have fixed that with 1.11.2 version. You have to uninstall and reinstall and it should be good.
Then, if you want to edit the black list, you can update your intunewin package with the new black list, or just deploy the black list file in the installation folder (by default c:\programdata\winget-autoupdate)

**Reply by mtengmo** [reply]
Thanks, similar to this deployment method with a xcopy/powershell script to copy the file to correct location?
https://www.cloudwisdom.co.uk/post/use-a-win32-app-to-copy-files-to-a-local-windows-device

Or is it an easier way?

**Reply by Romanitho** [reply]
yes, but you can use modification date as detection method :)

**Reply by jaredonair** [reply]
The issue I am running into is the logs are not detecting WingGet when the task runs. I have wrapped the entire folder, used the enclosed install.bat file. It installs successfully via intune.

If I look at the logs, it's not finding winget:

##################################################
#     CHECK FOR APP UPDATES - 9/15/2022
##################################################
14:51:13 - Notification Level: None. Notification Language: English
14:51:13 - Checking internet connection...
14:51:14 - Connected !
14:51:15 - Winget not installed or detected !
14:51:15 - End of process!

I CAN open powershell and run it manually and I will get success.

**Reply by jaredonair** [reply]
OK looks like it's an issue with the VM running on my apple M1. Works on 64 bit.

**Reply by Romanitho** [reply]
Should work on a VM :/

**Comment by jarskief** [comment]
Hi,
If you still have issues with deploying WAU with Intune this is how I do it and it works perfectly (100% success in installations without issues so far). 
- wrap entire folder with WAU using IntuneWinAppUtil 
- deploy .intunewin app from Intune with: PowerShell.exe -windowstyle hidden -ExecutionPolicy Bypass -File .\Winget-AutoUpdate-Install.ps1
And for the detection rules I use: File or folder in path c:\ProgramData\Winget-AutoUpdate\ (just pick one)
/J

**Reply by Romanitho** [reply]
Be careful, because in that way it will put the registry keys in the wow64 thing. I would suggest something like this:  
`"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -noprofile -executionpolicy bypass -file "Winget-AutoUpdate-Install.ps1" -Silent`
With the `silent` switch
Then, as detection rule, I would go for the registry, that will confirm it is perfectly installed in the right place.
`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate` and select the DisplayVersion than should be greater or equal to the version you try to deploy.
Not yet documented. Very busy at the moment :/

**Comment by DarkKnight-6-6-6** [comment]
hi Guys, 
I try to deploy it via Intune to device. But is stuck on "Pending Install"
![Screenshot 2022-10-10 150341](https://user-images.githubusercontent.com/115480103/194884412-47fc44c0-4659-4e7b-9964-5e38b5a71298.png)

**Reply by DarkKnight-6-6-6** [reply]
something need to be change?

**Reply by Romanitho** [reply]
It seams good. This is what I have on one of my customers:

![image](https://user-images.githubusercontent.com/96626929/194896102-ce6155cd-80cd-48d8-9213-2540401cf265.png)

**Reply by DarkKnight-6-6-6** [reply]
Thanks for info. 
Will wait maybe Intune is slow :)

**Reply by KnifMelti** [reply]
-Uninstal should be -Uninstall (but that's when uninstalling).
1 Retry...
20H2 as minimum...

**Reply by KnifMelti** [reply]
1 as a Retry makes it miss if it Fails, or?

**Reply by DarkKnight-6-6-6** [reply]
hi Guy ,
quick question to disable notification level i just use -NotificationLevel None or None must be in ""

**Reply by KnifMelti** [reply]
Just None after the param: -NotificationLevel None

**Reply by DarkKnight-6-6-6** [reply]
Thanks

**Comment by jhochwald** [comment]
![Screenshot 2022-10-25 183814](https://user-images.githubusercontent.com/1397168/197832536-148e733f-1bd1-4b40-9840-56c761707897.png)
I decided to use the existing Batch Files within Intune. In this case, a newer version can be deployed without any changes to the App Installation/uninstallation, just change the files itself.
For the detection, I decided to use a script (instead of Intune registry detection). If you decide to make it more flexible and/or independent, you might want to use this very generic approach:
![Screenshot 2022-10-25 184305](https://user-images.githubusercontent.com/1397168/197833519-9144cd13-a27f-468b-94b6-2a66d98d2d6f.png)
In this case, Intune will no longer care about the installed version or anything else.

**Reply by Romanitho** [reply]
the thing is, all the registry keys will go to the WOW6432Node. And things might not work as expected.

**Reply by iMonZ** [reply]
> ![Screenshot 2022-10-25 183814](https://user-images.githubusercontent.com/1397168/197832536-148e733f-1bd1-4b40-9840-56c761707897.png) I decided to use the existing Batch Files within Intune. In this case, a newer version can be deployed without any changes to the App Installation/uninstallation, just change the files itself. For the detection, I decided to use a script (instead of Intune registry detection). If you decide to make it more flexible and/or independent, you might want to use this very generic approach: ![Screenshot 2022-10-25 184305](https://user-images.githubusercontent.com/1397168/197833519-9144cd13-a27f-468b-94b6-2a66d98d2d6f.png) In this case, Intune will no longer care about the installed version or anything else.

Are you sure it's not "winget-upgrade.ps1" (case-sensitive) at the detection method?

**Reply by Romanitho** [reply]
no case sensitive. just problem between 32/64 bits

**Reply by juanpalin90** [reply]
How do you run the install.bat it generates error 0x80070000 for me, when running the .bat locally it does not generate any action on my machine either, what should I adjust in the bat so that I can run it intune?
![image](https://user-images.githubusercontent.com/125480051/219798773-5073f675-f3cc-4f7e-9140-e6a841a59b1c.png)

**Reply by KnifMelti** [reply]
Hello!
Could you provide the output from cmd when running it locally for clarity, thanks!

**Reply by KnifMelti** [reply]
Ah, you've got spaces before and after the equal sign...
,,,this is the correct:
`SET PowershellCmd=Start-Process`

**Reply by juanpalin90** [reply]
Yes, notepad++ does not have space, I adjust it like this, but in notepad it looks without spaces.
when opening the .bat it does not generate errors, it only opens and closes a PW and does not do any action, it does not create any folder in programData

![image](https://user-images.githubusercontent.com/125480051/220209052-ab33b32b-f71a-4ca9-9f46-648d11e13a87.png)

**Reply by KnifMelti** [reply]
Well...
...if you then edit the last SET command to (it will keep the output in the command window):
```
SET PowershellCmd=Start-Process powershell.exe -NoNewWindow -Wait -Argument '-noprofile -executionpolicy bypass -file "%~dp0Winget-AutoUpdate-Install.ps1" %arguments%
```
- Start a command window (Run as administrator)
- Drag the install.bat to the command window
- Press Enter in the command window

That will show you what's the problem, for example if the **install.bat** runs from a path with spaces in it there will be an error (in my case down below from **New folder**):
```
C:\Users\WDAGUtilityAccount>"C:\Users\WDAGUtilityAccount\Desktop\New folder\install.bat"
Processing -File 'C:\Users\WDAGUtilityAccount\Desktop\New' failed because the file does not have a '.ps1' extension. Specify a valid Windows PowerShell script file name, and then try again.
```

**Reply by KnifMelti** [reply]
HowTo install it in Intune (up above):
https://github.com/Romanitho/Winget-AutoUpdate/discussions/88#discussioncomment-2849475

**Reply by juanpalin90** [reply]
Hello, I managed to run the .bat with version 1.15.3, it worked for me, but I still can't get it recognized by Intune, I have two questions.
1. What are the files left to pack with IntuneWinAppUtil.exe
2. To execute the install.bat, what command line should I place in the intune configuration, since if I only adjust install.bat, the task will not be executed. Should I put a cmd command line that opens the .zip and looks for the install.bat to execute the task on the assigned computers?
![image](https://user-images.githubusercontent.com/125480051/220703728-64d9fc28-8f09-4502-a57c-305ce4ce06d9.png)

**Reply by KnifMelti** [reply]
Still the best way is https://github.com/Romanitho/Winget-AutoUpdate/discussions/88#discussioncomment-2849475 (I don't work with Intune a lot, SCCM is my game) with your chosen parameters:

Install cmd (for example with specific notification level):
"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -noprofile -executionpolicy bypass -file "Winget-AutoUpdate-Install.ps1" -Silent -NotificationLevel SuccessOnly

Uninstall cmd:
"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -noprofile -executionpolicy bypass -file "Winget-AutoUpdate-Install.ps1" -Uninstall

Detection method (with Version value for the one you're installing:
Key path: HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate
![image](https://user-images.githubusercontent.com/102996177/220705374-34a11597-712a-48c4-b225-1d6b29a4ef85.png)

**Comment by Tionic** [comment]
Hi, I am getting a windows error 0x80070001 when installing via the company portal on my test environment. Below is my configuration on intune - 
![image](https://user-images.githubusercontent.com/121228410/209115200-3850d1d2-203c-4455-a6d6-a99e2b77784b.png)
Wondering if anyone else has had this issue or if i'm missing something in the setup - Thanks

**Reply by KnifMelti** [reply]
Error 0x80070001 seems to be a lot...
Have you tried rebooting the client?
Try making a **Win32 application** instead:
https://github.com/Romanitho/Winget-AutoUpdate/discussions/88#discussioncomment-2849475

**Reply by Romanitho** [reply]
what do you have inside your intunewin ?

**Reply by TdunkProg** [reply]
Having the same issue. I repackaged everything in the main folder and set the WInget-AutoUpdate.ps1 as the installer in the win32 repackager

**Comment by Banjo-sys** [comment]
Hey @Romanitho,

when we install this script via Intune everything works perfect except the notification, its always the default without the name of the updated app.

Logs report the App that got updated do you have any clue what can cause this issue?

**Reply by Romanitho** [reply]
hard to say. Could you share few screenshots of the notifs, your intune config, or everything you think that could help ? thanks ;)

**Reply by ak47uk** [reply]
Could it be related to this? https://github.com/Romanitho/Winget-AutoUpdate/issues/276

**Comment by lukecoleman007** [comment]
I have installed this via intune and it runs on my test machine, but when it tries to update anything they just fail , have tried as a standard user and as an admin account, the results are the same 
any help greatly appreciated!
![Greenshot 2023-03-16 14 39 51](https://user-images.githubusercontent.com/112495335/225652162-241646d9-5f25-4bf0-8a23-5e84da2fc904.png)

**Reply by Romanitho** [reply]
Hi,
Do you have log file to share with us on the computer it fails ?

**Comment by lukecoleman007** [comment]
Sure, here it is.
on a side note, it looks like it did update in the end , but still says it failed?




Luke Coleman
Identity & Access Management Engineer

[Graphical user interface, text, application  Description automatically generated]


[AXGjCT8RZH54QAAAABJRU5ErkJggg==]

Zilch Technology Limited
123 Buckingham Palace Road | London | SW1W 9SH
***@***.******@***.***>

Follow us on LinkedIn<http://bit.ly/3O9t06f> for our latest updates


From: Romain ***@***.***>
Date: Friday, 17 March 2023 at 08:47
To: Romanitho/Winget-AutoUpdate ***@***.***>
Cc: Luke Coleman ***@***.***>, Comment ***@***.***>
Subject: Re: [Romanitho/Winget-AutoUpdate] Intune Deployment (Discussion #88)

Hi,
Do you have log file to share with us on the computer it fails ?

**Comment by TdunkProg** [comment]
Would you be willing to help me set this up for Education and with msstore integration? I'll need to remove the azure blob storage part as it has some of your info in it to help schools. Let me know. tyler.duncan@outlook.com

**Comment by marvnl** [comment]
Not sure if anyone here has the right setup for this already, but the folder "%systemroot%\sysnative\WindowsPowerShell\v1.0\" is not to be found on our devices. Instead, it is "%systemroot%\WinSxS\wow64_microsoft-windows-powershell-exe_31bf3856ad364e35_10.0.22621.1_none_d50074ba2a5195be\". 

But you can also do as install and uninstall command after repacking it with the powershell  script "Winget-AutoUpdate-Install.ps1" as file:

Install:
powershell -executionpolicy bypass -file Winget-AutoUpdate-Install.ps1 -install

Uninstall:
powershell.exe -noprofile -executionpolicy bypass -file "C:\ProgramData\Winget-AutoUpdate\WAU-Uninstall.ps1"

**Reply by Romanitho** [reply]
Are you deploying for user or system ?

**Reply by marvnl** [reply]
As system deploy.

**Reply by Romanitho** [reply]
Weird 🤔

**Reply by Erdnuckl** [reply]
I did some investigations in the last two days. It looks to me, that the "restart as 64bit" function is the problem. I created a start script which creates some install logs. I used a different script from the web to start the "Winget-AutoUpdate-Install" already as 64bit. In addition to that I need to start the script from Intune like that: 

powershell.exe -ExecutionPolicy Bypass -File .\Start-Install.ps1

**Reply by Romanitho** [reply]
`powershell.exe -ExecutionPolicy Bypass -File .\Start-Install.ps1` will run powershell as 32bit process in Intune.

**Reply by Erdnuckl** [reply]
Now I'm starting to get curious because this whole Intune PS 32bit vs 64bit is new to me. Therefore I did some additional testing:

Test1:
Parameter: powershell -executionpolicy bypass -file test64.ps1 
Result: $env:PROCESSOR_ARCHITECTURE = x86

Test2:
Parameter: powershell.exe -ExecutionPolicy Bypass -File .\test32.ps1
Result: $env:PROCESSOR_ARCHITECTURE = x86

I did both test on:
Microsoft Windows NT 10.0.22000.0
PSVersion: 5.1.22000.1335
PSEdition: Desktop
PSCompatibleVersions: 1.0, 2.0, 3.0, 4.0, 5.0, 5.1.22000.1335
BuildVersion: 10.0.22000.1335
CLRVersion: 4.0.30319.42000
WSManStackVersion: 3.0
PSRemotingProtocolVersion: 2.3
SerializationVersion: 1.1.0.1

**Reply by Romanitho** [reply]
Intune Win32 app runs under 32 bits process, so it starts PowerShell in 32 bits. That's why we call "%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe"

**Comment by DarkKnight-6-6-6** [comment]
hi guys if its possible to add lists of app to iinstall ?

**Reply by Romanitho** [reply]
Yes, just put your included_list before packaging the intunewin

**Comment by MZ-AI** [comment]
Hi, it seems your current versioning isn't compatible with intune. Would it be possible to get back to somthing like 1.7.8.3 for future releases?
![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/66481053/14319260-103c-4bb6-a87c-7d82812eb170)

**Reply by Romanitho** [reply]
We do not recommend using pre-release version in enterprise environment. Secondly, we cannot get back to previous release versioning due to automation. We use GitHub action to automate the releases.
If you still want to deploy this version anyway, just manually change the version in the `Winget-AutoUpdate/Version.txt` and disable WAU self update.

**Reply by KnifMelti** [reply]
Or create another detection (I inserts the date of deployment into registry and then check that)

**Reply by MZ-AI** [reply]
Using this on test devices atm. Obviously had to use another detection. Just saying a proper versioning would be great for **future** releases.

**Reply by dahennessy** [reply]
I install through a W32 app with this command:
`"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -noprofile -executionpolicy bypass -file "Winget-AutoUpdate-Install.ps1" -Silent -NotificationLevel SuccessOnly`
But on installation, still the powershell pops up and takes at least 10 minutes to update all apps, without any process information. For me, this is no issue at all but I can't use that with our employees. Am I doing something wrong or is the -Silent switch not working?

**Reply by Romanitho** [reply]
First, PowerShell should not pop up if installed as system.
Second, add a `-DoNotUpdate` argument to avoid instant updates. This way, WAU will run on next scheduled trigger.

**Comment by Dobbies-Abdale** [comment]
Has anyone managed to get this to work on the latest versions of Windows 10 or Windows 11? 

On Windows 11, the only way I could get the install to run from Intune deployment was to comment out the 32bit/64bit check and call powershell as normal because %systemroot%\sysnative\ doesn't exist. Install ran fine and was x64, but when it got to detecting and installing WinGet it failed. I think this is the same brick wall I've hit with other solutions where trying to run WinGet in the system context it looks at a different version from a different folder.


################################################################
#     18/01/2024 -   INSTALLING WAU
################################################################
17:40:02 - Checking prerequisites...
17:40:04 - Prerequisites checked. OK

17:40:04 - Checking if WinGet is installed/up to date.
17:40:04 - -> WinGet is not installed
17:40:05 - -> Downloading WinGet MSIXBundle for App Installer...
17:40:14 - -> Installing WinGet MSIXBundle for App Installer...
17:40:19 - -> WinGet MSIXBundle (v1.6.3482) for App Installer installed successfully!
17:40:19 - -> Failed to install WinGet MSIXBundle for App Installer...

17:40:19 - -> Forcing an upgrade of Store Apps...
17:40:20 - -> ...this can take a minute!
17:42:02 - Winget is mandatory to execute WAU.
17:42:02 - End of process.

**Reply by modernendpointde** [reply]
Same error here

**Reply by Dobbies-Abdale** [reply]
I got it working by separately deploying WinGet to each PC so I knew it was installed, then commenting out the Update-WinGet and Update-StoreApps functions (as well as the x32/x64 check) in the Winget-AutoUpdate-Install script. It then deployed perfectly via Intune and is updating WAU and apps as expected on Windows 11.

**Reply by modernendpointde** [reply]
Thank you! I was also planning to proceed by commenting out parts of the code. We have already installed Winget through a different script. The self-update feature of Autoupdate has always worked until now. It's possible that this issue is related to the new Winget version released in December.
Still using Windows 10.

**Comment by piggersyo** [comment]
Can a particular day be chosen in the scripts?
For example if I install today - the script will run next week.
Can I set for example to run Every Monday weekly?

**Reply by Romanitho** [reply]
by default it's Tuesday. But it can be set differently in the scripts. we probably need to adapt this.

**Reply by Dobbies-Abdale** [reply]
Yes, check the Winget-AutoUpdate-Install.ps1 script for the available parameters:

[Parameter(Mandatory = $False)] [ValidateSet("Daily", "BiDaily", "Weekly", "BiWeekly", "Monthly", "Never")] [String] $UpdatesInterval = "Daily",
[Parameter(Mandatory = $False)] [DateTime] $UpdatesAtTime = ("06am"),

**Reply by piggersyo** [reply]
You can see "Weekly" but can you select a difference day than Tuesday

**Comment by piggersyo** [comment]
![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/164195164/f8ba728d-bbd9-45f6-8e1a-651ff951f72b)

So - here you can see 6:00am and 11am weekly.
Is the 6am to update the software? or what is is for

**Reply by Romanitho** [reply]
it is for policies, if WAU is used with GPO in enterprise context. If not used, the scheduled task does nothing.

**Comment by eduhazard** [comment]
I'm trying to use ADMX to manage it on Intune, and the goal is to blacklist a few apps which keep failing or appear as updated plus hide the notifications etc.
![2024-03-28_15h54_16](https://github.com/Romanitho/Winget-AutoUpdate/assets/58125192/10e6c2db-ce8e-4a58-a65b-a099597ad6da)

For that, I enabled "Application GPO Blacklist" and added some App's IDs to the list. Running it on my machine as a test, I found the policy 
successful on Intune. But I want to know how to verify if the changes I made are updated on the machine. 
All the reg keys seem the same as before the change (HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate)

Thoughts?

**Reply by KnifMelti** [reply]
https://github.com/Romanitho/Winget-AutoUpdate/blob/3b8d1f5ff26cf5bc543a7b1364689fe7bf545442/Sources/WAU/Winget-AutoUpdate/functions/Get-ExcludedApps.ps1
HKLM:\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate\BlackList

**Reply by eduhazard** [reply]
it worked!

![2024-04-02_07h51_41](https://github.com/Romanitho/Winget-AutoUpdate/assets/58125192/d6d818b2-c580-408a-974e-c4ab5937eed5)

thanks

**Comment by eduhazard** [comment]
> https://github.com/Romanitho/Winget-AutoUpdate/blob/3b8d1f5ff26cf5bc543a7b1364689fe7bf545442/Sources/WAU/Winget-AutoUpdate/functions/Get-ExcludedApps.ps1 HKLM:\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate\BlackList

@KnifMelti, in the ps1 there's a line mentioning "excluded_apps.txt". Do I also need to update that TXT? Or are the reg changes(from Intune) enough?

**Reply by KnifMelti** [reply]
To use GPO as an external path for the lists (fetched from elsewhere than directly from where WAU is installed) you must (as you've already done) enable and fill the list in GPO via:

- **Application GPO Blacklist**

To make WAU use the GPO list you must enable and edit the:

- **Get Black/White List from external Path (URL/UNC/GPO/Local)**

For GPO the **(URL/UNC/GPO/Local) Path** must be:
**GPO**
![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/102996177/4675485c-6d35-476c-8367-81b5e7a061af)

In this way you can shift between different external paths without loosing the lists in GPO (as will happen if you disable the **Application GPO Blacklist**)

**Reply by eduhazard** [reply]
Great, so to have the blocklist working through Intune, I must set up, as you said, an external path given, let's say, a URL providing a TXT file with all the AppIDs I want in the blocklist.
![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/58125192/b7bbee0c-024b-4f75-8347-57b8e98d5c6f)

I enabled "Get Black/White List from external Path (URL/UNC/GPO/Local)" on Intune's policy and added a URL linked to it, file as below:
![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/58125192/66d873dd-06c8-4095-a15d-9326e76efbbf)

So, do you think it should work now?

Another question, as you mentioned in the last, swift and disable "Application GPO Blacklist" which I started this discussion. 
Don't we need to keep both lists of apps in both locations? Application GPO Blacklist and External Path URL? 
As I understood, it seems External Path is the "main", but I want to confirm because I can't see what's the purpose of "Application GPO Blacklist" if we still need to set another external list to have it working.

I don't know if it matters, but we're using Intune only to apply all our GPOs, policies etc.

Plus, thanks a lot for the quick response! Appreciate that, take your time if you're busy.

**Reply by KnifMelti** [reply]
If you're planning to use a Path for the list:
- an URL (like `https://www.knifmelti.se/wau/lists`)
- an UNC (like `\\someshare\somewhere`)
- a local Path (like `E:\WAU\lists`)
- the file `excluded_apps.txt` or `included_apps.txt` (depending on what you will use) under that Path

That external file in the Path will be date/time compared to the local under WAU installation existing `excluded_apps.txt` or `included_apps.txt`
If newer or if missing under WAU installation the file will be downloaded and used by WAU

If you're planning to use GPO for the list:
- the word `GPO`
- fill in the list in **Application GPO Black/Whitelist**

The GPO list is used internally by WAU without any file

**Comment by AFJ0** [comment]
Hello everyone, been lurking for a while but I think it's time to ask for some guidance I simply can't understand what I'm doing wrong.

I'm trying to deploy WAU to keep other applications that we're deploying through intune up to date.

As far as what I've been able to read in this thread, I would say that I've configured the setup for deployment correctly but when trying to install WAU either as required or available it simply fails. So obviously I'm missing something.

Checking the Device install status in Intune the "Status detail" shows **0x80070000**. If it's possible for me to somehow provide a more indepth log as to why it fails, let me know and I'll try to provide it.

Below is the process I've tried to follow aswell as the configurations I've set up in Intune.

I've downloaded the Winget-AutoUpdate-Install.ps1 and creating the .Intunewin using the [Intune Prep tool](https://github.com/Microsoft/Microsoft-Win32-Content-Prep-Tool). Other solutions in this thread https://github.com/Romanitho/Winget-AutoUpdate/discussions/88#discussioncomment-3660900 is to download the repository and and use the prep tool. I've tried this and it fails aswell.

![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/74936543/ee0a2a6d-0aa9-46a8-a96c-89e29afdd25a)

![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/74936543/23941d07-c906-4e38-82a0-aeebd8b065d3)

![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/74936543/00f477fa-3362-4581-a733-2f134925b959)

**Reply by piggersyo** [reply]
Hey Buddy!

Happy to help you get this setup.

Here is my detection method.

HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate

![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/164195164/23603f19-9477-419a-9ffb-bf7344c3d8f0)

Install Command Line

"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -noprofile -executionpolicy bypass -file "install.ps1" -Silent

Uninstall Command Line

"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -noprofile -executionpolicy bypass -file "Uninstall.ps1" -Silent

Mine works really well - if you need a hand just let me know.

![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/164195164/c98da20e-0e03-48ed-accd-e1d9112bbfa3)

**Reply by AFJ0** [reply]
Thanks for the quick response.
I've tried changing the settings in in my configuration to match yours and the installation is still failing.

Just for good measure I cloned the WAU repo again and re-reprepped the "Winget-AutoUpdate-Install.ps1" script and reuploaded the .intunewin file
I'm still fairly new to application deployment through Intune, so I'm unsure if I'm missing important information as to why this might fail. We have multiple win32 apps deployed using powershell similar to what I'm trying to accomplish here.

**Reply by piggersyo** [reply]
Hey Buddy!

You can have my files - Folder contains the Intunewin file etc
https://cloudperformanceuk.sharepoint.com/:f:/s/CloudPerformanceConfidential/EuJChcY6NE1NjjXvryc9N7EBJLtjjdwIRqXv44QZoc_9HQ?e=yQ2gnf

Hope this helps!

**Reply by AFJ0** [reply]
Thanks.

I'll take a look at it and post an update when I've reached a conclusion.

**Reply by abrahamreyes9** [reply]
@piggersyo 

Could you share the files with me also I'm also stuck

**Reply by piggersyo** [reply]
Sure 

https://cloudperformanceuk.sharepoint.com/:f:/s/CloudPerformanceConfidential/EuJChcY6NE1NjjXvryc9N7EBJLtjjdwIRqXv44QZoc_9HQ?e=dBvTVc

**Reply by TomCD97** [reply]
@piggersyo sorry mate could you share the files also with me? This is how i'm packaging the installer - 
![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/67586869/3b3e4651-9d48-4917-a875-b903159bc450)

**Comment by ThomasVIDALUBS** [comment]
Hi there,

I tried all solutions / path in comments .... but WAU failed to install with intune without comprehensive log or message...
We're using Win11 devices.
Is there any chance to get this great tool installing with Win32 apps in intune ?

Many thx

T.

**Reply by Romanitho** [reply]
you need to bring more details on the way you do it.

**Reply by ThomasVIDALUBS** [reply]
Hi Romanitho,

I'm creating the intunewin package as usual : 
![image](https://github.com/user-attachments/assets/cec1c5cf-59aa-4b23-9cd7-faf063063181)

Then the last try of the intune config is : 

![image](https://github.com/user-attachments/assets/5e011b75-63bf-4511-a96d-800080bd811e)

Many thanks !!

**Reply by Romanitho** [reply]
have you tried with sysnative powershell? Check other comments ;)

**Comment by bmacks1** [comment]
Hoping to get some insight and a little guidance with getting applications to install when users are not Admins. 

I was able to deploy this within Intune using the following commands. I did try all those provided in the above posts, but this was the only method that would deploy correctly. I would consider myself a novice with PowerShell so I may be missing something. 

06:00:05 - Winget Version: v1.8.1911
06:00:05 - WAU current version: 1.20.2
06:00:05 - WAU AutoUpdate is Disabled.
06:00:05 - WAU uses Black List config

![image](https://github.com/user-attachments/assets/da15f137-5feb-4131-95c4-5082dcec31f2)

The main issue I am running into is WinGet failing to install "SOME" application updates due to end users not being an admin on the machine. I am not sure if this is linked to the Task Scheduler for "Winget-AutoUpdate-UserContext" and wanted to see if there is any way to get a service account that has admin permissions to update the applications. 

It looks like the failures start after -Starting WAU in User Context... runs. No additional logs are generated following this.

A few of the items failing are Dev Home, Windows Terminal, Microsoft Teams (Work or School).  

Thank you,

EDIT: This may be an issue with applications from the Microsoft Store and not an admin issue. Can you point the application to multiple repositories?

**Reply by incedIT** [reply]
Dev Home is causing me issues too so maybe you’re just experiencing it with some apps that require UAC. In my case, I have set WAU to ignore Dev Home and added it as a winget package to Intune apps and set to uninstall on all devices - I don’t want it and not sure why MS pushed it on everyone. 

I think Teams was recommended to go on the do not update list, along with web browsers as they all have internal mechs to update regularly so no need for WAU to do it.
